### ISSUE TITLE
`query xml` only finds not-namespaced tags

### ISSUE THREAD

**Issue by Kissaki** [nushell/nushell#14457]
### Describe the bug

With `query xml` I can query XML by XPath.

When tags are namespaced querying becomes a hassle. Namespaces can be queried with `namespace::alias`, but that only works when they are named tag namespaces.

### How to reproduce

```
nu ❯ '<?xml version="1.0" encoding="utf-8"?><a xmlns="http://example.org/any"><b>test</b></a>' | query xml '//b'
╭────────────╮
│ empty list │
╰────────────╯

nu ❯ '<?xml version="1.0" encoding="utf-8"?><a xmlns:n="http://example.org/any"><b>test</b></a>' | query xml '//b'
╭───┬──────╮
│ # │ //b  │
├───┼──────┤
│ 0 │ test │
╰───┴──────╯

nu ❯ '<?xml version="1.0" encoding="utf-8"?><a xmlns:n="http://example.org/any"><b>test</b></a>' | query xml '//b[namespace::n]'
╭───┬───────────────────╮
│ # │ //b[namespace::n] │
├───┼───────────────────┤
│ 0 │ test              │
╰───┴───────────────────╯

nu ❯ '<?xml version="1.0" encoding="utf-8"?><a xmlns="http://example.org/any"><b>test</b></a>' | query xml '//b[namespace::*]'
╭────────────╮
│ empty list │
╰────────────╯
```

### Expected behavior

I would expect `query xml '//b'` to return tags of any namespace, not only those not in any namespace.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.100.0                                  |
| major              | 0                                        |
| minor              | 100                                      |
| patch              | 0                                        |
| branch             |                                          |
| commit_hash        | 30f98f7e64a63a159ce70f02aa6c4c1196cd9ca7 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06)      |
| rust_channel       | 1.80.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.80.1 (376290515 2024-07-16)      |
| build_time         | 2024-11-13 03:41:33 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash                   |
| installed_plugins  | query 0.100.0                            |

**Comment by Kissaki** [comment]
There is no unit test for this in https://github.com/nushell/nushell/blob/main/crates/nu_plugin_query/src/query_xml.rs

Related previously there was #8624

**Comment by Kissaki** [comment]
Workaround:

```
 | str replace 'xmlns="http://example.org/any"' ''
```

**Comment by fdncred** [comment]
Definitely would take a PR for fixing up things like this. I've noticed a few times that query xml didn't do what I wanted and acted a bit weird.

**Comment by weirdan** [comment]
> I would expect `query xml '//b'` to return tags of any namespace, 

I don't think this is how namespaces are supposed to work in XML. Instead, most libraries allow you to specify mapping between prefixes (which do not have to match those used in the document) and namespace URIs and use those prefixes to query the XML.

To mimic this in `query xml` I'd add `--namespace {ns1: "http://....", prefix: "http://...", ...etc...}` instead

**Comment by fdncred** [comment]
closed by https://github.com/nushell/nushell/pull/16008
